### PR TITLE
ExtensionMethods update

### PR DESCRIPTION
**Pull request by LingZhou-gh**
Minor (not significant) change
Add some extension functions to check whether a string contains digits and/or letters.

### CODE REVIEW

File: Src/Plugins/Security/SecurePlaintextSecretsValidators/SEC101_006.GitHubLegacyPatValidator.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ protected override IEnumerable<ValidationResult> IsValidStaticHelper(IDictionary
 
             // It is highly likely we do not have a key if we can't
             // find at least one letter and digit within the pattern.
-            if (!ContainsDigitAndChar(secret.Value))
+            if (!secret.Value.ToString().ContainsDigitAndLetter())
```
Comment (michaelcfanning) [inline Src/Plugins/Security/SecurePlaintextSecretsValidators/SEC101_006.GitHubLegacyPatValidator.cs:42]: >ToString() [](http://example.com/codeflow?start=30&length=10)

are these ToString() calls actually necessary?

I think secret.Value is already a string.

**Review by michaelcfanning — APPROVED** [review]
:shipit:

File: Src/Plugins/Security/SecurePlaintextSecretsValidators/SEC101_006.GitHubLegacyPatValidator.cs (hunk 1 above)
Comment (LingZhou-gh) [reply Src/Plugins/Security/SecurePlaintextSecretsValidators/SEC101_006.GitHubLegacyPatValidator.cs:42]: I thought the same thing before. 
secret.Value is FlexString. 
secret.Value.ContainsDigitAndLetter() won't work here.

Comment (LingZhou-gh) [reply Src/Plugins/Security/SecurePlaintextSecretsValidators/SEC101_006.GitHubLegacyPatValidator.cs:42]: image

**Pull request merged**